src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] values;
+    private int size;
 
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_SIZE];
     }
 
-    @Override
-    public void add(T value, int index) {
+    public ArrayList(int initialCapacity) {
+        values = (T[]) new Object[initialCapacity];
+    }
 
+    @Override
+    public int size() {
+        return size;
     }
 
     @Override
-    public void addAll(List<T> list) {
+    public void add(T object) {
+        resize();
+        values[size++] = object;
+    }
 
+    @Override
+    public void add(T object, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound!");
+        }
+        resize();
+        System.arraycopy(values, index, values, index + 1, size - index);
+        values[index] = object;
+        ++size;
     }
 
     @Override
-    public T get(int index) {
-        return null;
+    public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
-    public void set(T value, int index) {
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    public T get(int index) {
+        checkInBounds(index);
+        return (T) values[index];
+    }
 
+    public int indexOf(T value) {
+        for (int i = 0; i < size; i++) {
+            if (values[i] == value || (values[i] != null && values[i].equals(value))) {
+                return i;
+            }
+        }
+        return -1;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkInBounds(index);
+        T removedObject = values[index];
+        System.arraycopy(values, index + 1, values, index, size - index - 1);
+        values[--size] = null;
+        return removedObject;
     }
 
     @Override
-    public T remove(T element) {
-        return null;
+    public T remove(T value) {
+        int index = indexOf(value);
+        if (index == -1) {
+            throw new NoSuchElementException("There is no element like this!");
+        }
+        checkInBounds(index);
```

Comment:
When you find the index - use the method `remove(int index)`. This method contains a check index mechanism